### PR TITLE
Feature/pointdetails

### PR DESCRIPTION
**Pull request by RtAnto**
# Description
Added front end functionality for viewing point details

## Type of change

- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

manually tested

### CODE REVIEW

File: Backend/WebApi-PointMap/web.config [hunk 1]
```diff
@@ -44,11 +44,15 @@
       <compiler language="vb;vbs;visualbasic;vbscript" extension=".vb" type="Microsoft.CodeDom.Providers.DotNetCompilerPlatform.VBCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" warningLevel="4" compilerOptions="/langversion:default /nowarn:41008 /define:_MYTYPE=\&quot;Web\&quot; /optionInfer+" />
     </compilers>
```
Comment (ChrisMeyer7088) [inline Backend/WebApi-PointMap/web.config:45]: What are the purpose of these settings you are adding?

File: Backend/UnitTesting/App.config [hunk 2]
```diff
@@ -1,25 +1,40 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ChrisMeyer7088) [inline Backend/UnitTesting/App.config:1]: Not sure why there is an App.config file being used, should already have an app.config file with this layout

File: Frontend/src/services/pointServices.js [hunk 3]
```diff
@@ -35,7 +35,31 @@ function getPoints(minLng, maxLng, minLat, maxLat, callback){
     })
     return null;
 };
+function grabProm(fetchurl)
```
Comment (ChrisMeyer7088) [inline Frontend/src/services/pointServices.js]: no real point in separating this into another function

File: Frontend/src/services/pointServices.js [hunk 4]
```diff
@@ -35,7 +35,31 @@ function getPoints(minLng, maxLng, minLat, maxLat, callback){
...
     return null;
 };
+function grabProm(fetchurl)
+{
+    return axios.get(fetchurl);
+};
+function getPoint(pointId, callback){
+    let arr =[]
+    let urlString = 'https://api.pointmap.net/api/point/'+pointId
+    grabProm(urlString).then((response) => {
```
Comment (ChrisMeyer7088) [inline Frontend/src/services/pointServices.js]: Will need to add a token header to authenticate your request, token should already be stored in local storage so like...
let token = localStorage.getItem("token")
something like that, then pass in token as a header for the request

File: Frontend/src/views/MapView.vue [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
 </template>
 
 <script>
 import {getPoints} from '../services/pointServices'
 import {gmapApi} from 'vue2-google-maps'
 import {checkSession} from '../services/authorizationService'
-
+import {PointDetails} from '../views/PointDetails'
 export default {
+  components: {PointDetails},
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/MapView.vue]: what is this line doing?

Comment (ChrisMeyer7088) [inline Frontend/src/views/MapView.vue]: what is this line doing?

File: Frontend/src/views/PointDetails.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      long: null,
+      lat: null
+
+    }
+  }, 
+  methods: {
+     getPointDetails: function()
+     { 
+         this.url = window.location.href
+         this.id = this.url.substring(this.url.length-36,this.url.length)
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/PointDetails.vue]: there is an easy way to get a header value of a route parameter by using
this.$route.query.pointId

File: Frontend/src/views/PointDetails.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
+<template>
+
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/PointDetails.vue:2]: Will need to run demo this after changes are done

**Review by ChrisMeyer7088 — CHANGES_REQUESTED** [review]

File: Backend/WebApi-PointMap/web.config (hunk 1 above)
Comment (RtAnto) [reply Backend/WebApi-PointMap/web.config:45]: No idea, it seemed like it was automatically changed, I didn't manually change it

File: Frontend/src/views/MapView.vue (hunk 5 above)
Comment (RtAnto) [reply Frontend/src/views/MapView.vue]: I was trying to get both components on the same page, but decided just to make it a separate page for simplicity sake

File: Frontend/src/views/PointDetails.vue (hunk 6 above)
Comment (RtAnto) [reply Frontend/src/views/PointDetails.vue]: i will change this

File: Frontend/src/views/MapView.vue (hunk 5 above)
Comment (RtAnto) [reply Frontend/src/views/MapView.vue]: nothing, I will delete it

File: Frontend/src/services/pointServices.js (hunk 4 above)
Comment (RtAnto) [reply Frontend/src/services/pointServices.js]: i'll try this implementation

Comment (RtAnto) [reply Frontend/src/services/pointServices.js]: shouldn't I just call check session() from authorization services, I have it called already when the point details page is initially loaded.

Comment (ChrisMeyer7088) [reply Frontend/src/services/pointServices.js]: Every request to the backend should check your session including the request where you are fetching point information.
The checkSession services is a route thats called on a page launch to ensure no unauthorized user is accessing your page

File: Backend/WebApi-PointMap/web.config (hunk 1 above)
Comment (ChrisMeyer7088) [reply Backend/WebApi-PointMap/web.config:45]: please change it back to original if you are not sure why those settings are there, "git pull origin master" might help if your branch is not up to date with master

File: Frontend/src/views/MapView.vue [hunk 8]
```diff
@@ -1,15 +1,18 @@
 <template>
   <div>
-    <div v-on:click="requestPoints" id="map"></div>
+   <div><PointDetails></PointDetails></div>
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/MapView.vue]: Please change this back too

**Review by ChrisMeyer7088 — CHANGES_REQUESTED** [review]

**Review by ChrisMeyer7088 — APPROVED** [review]

**Pull request merged**